core/src/main/java/org/radargun/config/Cluster.java

```diff
@@ -53,23 +53,27 @@ public int getSize() {
    }
 
    public Group getGroup(int slaveIndex) {
-      int index = slaveIndex;
-      for (Group g : groups) {
-         if (index < g.size) {
-            return g;
+      if (slaveIndex != -1) {
```

Comment:
I'm not sure I agree. `-1` is the value used when no index has been assigned, so I think it is correct to check for that value, unless it is possible that some other value can be set?